examples/flower-secure-aggregation/secaggexample/client_app.py

```diff
@@ -0,0 +1,77 @@
+"""secaggexample: A Flower with SecAgg+ app."""
+
+import time
+
+import torch
+from secaggexample import task
+from secaggexample.task import Net, get_weights, load_data, set_weights, test, train
+
+from flwr.client import ClientApp, NumPyClient
+from flwr.client.mod import secaggplus_mod
+from flwr.common import Context
+
+
+# Define Flower Client
+class FlowerClient(NumPyClient):
+    def __init__(self, trainloader, valloader, local_epochs, learning_rate):
+        self.net = Net()
+        self.trainloader = trainloader
+        self.valloader = valloader
+        self.local_epochs = local_epochs
+        self.lr = learning_rate
+        self.device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+
+    def fit(self, parameters, config):
+        """Train the model with data of this client."""
+        set_weights(self.net, parameters)
+        results = train(
+            self.net,
+            self.trainloader,
+            self.valloader,
+            self.local_epochs,
+            self.lr,
+            self.device,
+        )
+        ret_vec = get_weights(self.net)
+
+        # Force a significant delay for testing purposes
+        if "drop" in config and config["drop"]:
+            print(f"Client dropped for testing purposes.")
+            time.sleep(15)
+        else:
+            print(f"Client uploading parameters: {ret_vec[0].flatten()[:3]}...")
+        return ret_vec, len(self.trainloader.dataset), results
+
+    def evaluate(self, parameters, config):
+        """Evaluate the model on the data this client has."""
+        set_weights(self.net, parameters)
+        loss, accuracy = test(self.net, self.valloader, self.device)
+        return loss, len(self.valloader.dataset), {"accuracy": accuracy}
+
+
+def client_fn(context: Context):
+    """Construct a Client that will be run in a ClientApp."""
+    # Set `task.is_demo`
+    task.is_demo = context.run_config["is-demo"]
```

Comment:
Would it be better to pass this `is_demo` to `load_data()` and then to the client so it skips calling `train()` and `test()`?

Reply:
The client has skipped `train` and `test` under the hood. In `task.py`, when `is_demo` is `True`, the `train` and `test` functions will directly return a fake result.